### PR TITLE
Chore(docs)i18n 404 by create pages

### LINKED ISSUE #23776: [docs] Add documentation of how to creating prefixed 404 pages by createPages
## Summary

I'm using `gatsby-plugin-i18n` to generate all pages dynamically with `createPages` API. When 404 error occurs, I'd like to redirect to the prefixed 404 pages for different languages. 
But after I ran into the [documentation](https://www.gatsbyjs.org/docs/creating-prefixed-404-pages-for-different-languages/), I found that there's only a sample of how to by using `onCreatePage` API.

### Motivation

Adding a sample of how to achieve the same goal by using `createPages` API is very kind for beginners implementing i18n by `gatsby-plugin-i18n`.

### Draft the doc

- [x] Write the doc, following the format listed in these resources:
  - [Overview on contributing to documentation](https://www.gatsbyjs.org/contributing/docs-contributions/)
  - [Docs Templates](https://www.gatsbyjs.org/contributing/docs-templates/)
  - [Example of a similar article]()
- [x] Add the article to the [docs sidebar](https://github.com/gatsbyjs/gatsby/blob/master/www/src/data/sidebars/doc-links.yaml) under the [parent doc] section.

### Open a pull request

- [x] Open a pull request with your work including the words "closes #[this issue's number]" in the pull request description

### PR DESCRIPTION
**Pull request by thundermiracle**
### Documentation

close #23776 add an example of how to redirect to prefixed 404 pages by createPages

### CODE REVIEW

**Comment by AishaBlake** [comment]
Since this is a reference guide, I'd also love to see some explanation as to why someone might choose one method over the other.

**Comment by thundermiracle** [comment]
@AishaBlake 
I added some details about the `createPages` flow. Please have a look.

>see some explanation as to why someone might choose one method over the other

This is little difficult for me. My only reason is that I'm using `gatsby-plugin-i18n` , want to create all pages dynamically and insist on DRY.
But it seems that 404 page is not that complex , so using `onCreatePage` and copy-paste is enough in my opinion.

File: docs/docs/creating-prefixed-404-pages-for-different-languages.md [hunk 1]
```diff
@@ -56,6 +58,96 @@ exports.onCreatePage = async ({ page, actions }) => {
...
+First of all, create a 404 template file in `src/templates/404.js`.
+
+```jsx:title=src/templates/404.js
+import React from "react";
+import { graphql, Link } from "gatsby";
+import Layout from "../../components/layout"
+
+export const query = graphql`
+  query NotFoundQuery($langKey: String!) {
+    markdownRemark(fields: { langKey: { eq: $langKey }, fileName: { regex: "/404/" } }) {
```
Comment (thundermiracle) [inline docs/docs/creating-prefixed-404-pages-for-different-languages.md:74]: @gillkyle 
First of all, this approach uses `gatsby-plugin-i18n` only in this graphql's filter and which is avoidable by filtering filename in render method without the plugin. 

But I still think that this approach is a little bit complicated for simple 404 pages.

As I described in #23776, what I were trying to do is using `createPages` API to create prefixed 404 pages but not `onCreatePage`.
So I searched `404` in docs and ran into [the documentation page](https://www.gatsbyjs.org/docs/creating-prefixed-404-pages-for-different-languages/). But unfortunately after I read the documentation, it seems that we can only do it with `onCreatePage` API like this.
```js
page.matchPath = `/${langCode}/*`
```

I don't want to use `onCreatePage` API as it never existed in my gatsby-node.js, so I tried a lot and finally found that if I pass `matchPath` to `createPage` all things work like a charm. I think someone should face the same problem of `how to use createPages to make prefixed pages?` no matter they're using i18n plugin or not. So I opened this issue here.

What about adding a simple `hint` for those people as my first commit did?
```
You can also do it by using the [`createPages`](/docs/node-apis/#createPages) API in your project's `gatsby-node.js` file to generate prefixed 404 pages dynamically and pass `langCode` to `matchPath`.

createPage({
    path: `/${langCode}/404/`,
    matchPath: `/${langCode}/*`,
    component: page404,
    context: {
      langCode,
    },
});
```

**Comment by gillkyle** [comment]
Hey there @thundermiracle, me again 👋  will try to respond to your comments:

The `gatsby-plugin-i18n` is mentioned in the overview page that precedes this doc so it isn't really worst thing to cover here. However, I still feel that improving the docs for plugin specific behavior in the plugin's README would be the ideal case. If the plugin is indeed only used in the graphql filter, then a plugin agnostic sample of code would probably serve a more general audience on this guide.

I think it's true that these docs could benefit from clearer example code. With a lot of existing plugins that implement things in different ways, i18n isn't one-size-fits-all. That being said, our guides should clear up as much confusion as possible and I'm hesitant that these changes might have an adverse effect of causing a reader to think they need the plugin, or should go down a different route than the other that's already outlined on the page with the `onCreatePage` API. I probably made a mistake turning this pull request into more of a discussion about how to address this rather than using the issue for that.

Moving forward, I believe it will help all parties if we close this pull request and carry on this discussion in the issue so that you don't have to keep changing code only to have us find a reason to tweak it again.

**Pull request closed without merging**